### PR TITLE
docs: Add website link and project description

### PR DESCRIPTION
**Pull request by georgebisbas**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,6 @@
-# xDSL
+# xDSL: A common ecosystem for DSL development
+
+[xDSL](http://www.xdsl.dev) is.....WIP WIP WIP
```
[the remark this answers is not in the document]
Comment (math-fehr) [reply README.md:3]: Yeah sorry, I'm a bit overwhelmed with paper writing currently, so I'll probably do it after the PLDI deadline if that's fine!

**Comment by georgebisbas** [comment]
@math-fehr just a gentle reminder for this, anytime you feel you have a few minutes

File: README.md (hunk 1 above)
Comment (webmiche) [inline README.md:3]:
```suggestion
[xDSL](http://www.xdsl.dev) is a compiler toolkit natively developed in Python. It applies the Multi-Level Rewrite Strategy, a compilation strategy that lowers programs through several, distinctly designed levels of abstraction. This allows the implementation of abstraction-specific optimization passes, similar to the structure of common DSL-compilers (such as Devito, Psyclone, an Firedrake). In xDSL, all data structures are SSA-based in order to simplify analysis and optimization passes within xDSL.

An inherent advantage of a design close to MLIR is the easy interaction among the two frameworks making it possible to translate abstractions back and forth. This results in one big SSA-based abstraction ecosystem that can be worked with through Python, making analysis through simple scripting languages possible. Additionally, xDSL can leverage MLIR's code generation and low-level optimization capabilities.
```

Comment (webmiche) [reply README.md:3]: I'm confused. Do we not want to commit this draft?

**Comment by webmiche** [comment]
@math-fehr @tobiasgrosser maybe give the description a quick read?

**Comment by math-fehr** [comment]
Nice! I added more text, feel free to comment on it!

**Comment by tobiasgrosser** [comment]
This looks great. Thank you!

**Pull request merged**